tests/utils/utils.ts

```diff
@@ -112,6 +112,13 @@ const cmdArgsArrayArb = fc
   })
   .noShrink();
 
+const vaultNameArb = fc
+  .stringOf(
+    fc.char().filter((c) => binParsers.vaultNameRegex.test(c)),
+    { minLength: 1, maxLength: 100 },
+  )
+  .noShrink();
```

Comment:
You should generate the string directly from the regex here.

Reply:
If I try to do this with `stringMatching`, then the tests fail to run with the following error:

```
Assertions of kind Lookahead not implemented yet!

      113 |   .noShrink();
      114 |
    > 115 | const vaultNameArb = fc.stringMatching(binParsers.vaultNameRegex).noShrink();
```

As such, I will retain the previous approach of checking individual characters and concatenating them.

Reply:
The previous approach looked very bad and also worked kinda weirdly. So, I have updated the code to utilise this instead of the previous approach.

```ts
const vaultNameArb = fc
  .string({ minLength: 1, maxLength: 100 })
  .filter((str) => binParsers.vaultNameRegex.test(str))
  .noShrink();
```